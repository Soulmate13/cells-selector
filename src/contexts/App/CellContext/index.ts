export type { Cell, StateType, ActionTypes } from './CellContext';
export { StateContext, DispatchContext } from './CellContext';
