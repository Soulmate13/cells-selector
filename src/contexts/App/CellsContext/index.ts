export type { Cell, StateType, ActionTypes } from './CellsContext';
export { StateContext, DispatchContext } from './CellsContext';
