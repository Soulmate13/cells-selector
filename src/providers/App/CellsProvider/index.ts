export { default, initialState } from './CellsProvider';
