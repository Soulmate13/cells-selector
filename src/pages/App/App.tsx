import { memo } from 'react';
import { ErrorBoundary } from 'react-error-boundary';
import ErrorFallback from 'components/organisms/ErrorFallback';
import Grid from 'components/organisms/Grid';
import CellsProvider from 'providers/App/CellsProvider';

const App = () => (
  <ErrorBoundary
    FallbackComponent={ErrorFallback}
    onReset={/* istanbul ignore next */() => window.location.reload()}
  >
    <CellsProvider>
      <div className="App">
        <h1 id="grid-heading">Cells Selector</h1>
        <Grid/>
      </div>
    </CellsProvider>
  </ErrorBoundary>

);

export default memo(App);
