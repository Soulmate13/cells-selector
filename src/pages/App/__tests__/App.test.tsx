import { render } from '@testing-library/react';
import App from 'pages/App';

describe('App', () => {
  describe('when initially rendered', () => {
    it('matches snapshot', () => {
      const { container } = render(<App/>);

      expect(container).toMatchSnapshot();
    });
  });
});
