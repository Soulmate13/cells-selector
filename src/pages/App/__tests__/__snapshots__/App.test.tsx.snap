// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App when initially rendered matches snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <h1
      id="grid-heading"
    >
      Cells Selector!
    </h1>
    <div
      aria-colcount="5"
      aria-describedby="grid-heading"
      class="grid"
      data-testid="grid"
      role="grid"
      tabindex="0"
    >
      <div
        class="row"
        role="row"
      >
        <button
          aria-colindex="1"
          aria-selected="false"
          class=" cell"
          data-testid="cell-1-1"
          role="gridcell"
        />
        <button
          aria-colindex="2"
          aria-selected="false"
          class=" cell"
          data-testid="cell-1-2"
          role="gridcell"
        />
        <button
          aria-colindex="3"
          aria-selected="false"
          class=" cell"
          data-testid="cell-1-3"
          role="gridcell"
        />
        <button
          aria-colindex="4"
          aria-selected="false"
          class=" cell"
          data-testid="cell-1-4"
          role="gridcell"
        />
        <button
          aria-colindex="5"
          aria-selected="false"
          class=" cell"
          data-testid="cell-1-5"
          role="gridcell"
        />
      </div>
      <div
        class="row"
        role="row"
      >
        <button
          aria-colindex="1"
          aria-selected="false"
          class=" cell"
          data-testid="cell-2-1"
          role="gridcell"
        />
        <button
          aria-colindex="2"
          aria-selected="false"
          class=" cell"
          data-testid="cell-2-2"
          role="gridcell"
        />
        <button
          aria-colindex="3"
          aria-selected="false"
          class=" cell"
          data-testid="cell-2-3"
          role="gridcell"
        />
        <button
          aria-colindex="4"
          aria-selected="false"
          class=" cell"
          data-testid="cell-2-4"
          role="gridcell"
        />
        <button
          aria-colindex="5"
          aria-selected="false"
          class=" cell"
          data-testid="cell-2-5"
          role="gridcell"
        />
      </div>
      <div
        class="row"
        role="row"
      >
        <button
          aria-colindex="1"
          aria-selected="false"
          class=" cell"
          data-testid="cell-3-1"
          role="gridcell"
        />
        <button
          aria-colindex="2"
          aria-selected="false"
          class=" cell"
          data-testid="cell-3-2"
          role="gridcell"
        />
        <button
          aria-colindex="3"
          aria-selected="false"
          class=" cell"
          data-testid="cell-3-3"
          role="gridcell"
        />
        <button
          aria-colindex="4"
          aria-selected="false"
          class=" cell"
          data-testid="cell-3-4"
          role="gridcell"
        />
        <button
          aria-colindex="5"
          aria-selected="false"
          class=" cell"
          data-testid="cell-3-5"
          role="gridcell"
        />
      </div>
      <div
        class="row"
        role="row"
      >
        <button
          aria-colindex="1"
          aria-selected="false"
          class=" cell"
          data-testid="cell-4-1"
          role="gridcell"
        />
        <button
          aria-colindex="2"
          aria-selected="false"
          class=" cell"
          data-testid="cell-4-2"
          role="gridcell"
        />
        <button
          aria-colindex="3"
          aria-selected="false"
          class=" cell"
          data-testid="cell-4-3"
          role="gridcell"
        />
        <button
          aria-colindex="4"
          aria-selected="false"
          class=" cell"
          data-testid="cell-4-4"
          role="gridcell"
        />
        <button
          aria-colindex="5"
          aria-selected="false"
          class=" cell"
          data-testid="cell-4-5"
          role="gridcell"
        />
      </div>
      <div
        class="row"
        role="row"
      >
        <button
          aria-colindex="1"
          aria-selected="false"
          class=" cell"
          data-testid="cell-5-1"
          role="gridcell"
        />
        <button
          aria-colindex="2"
          aria-selected="false"
          class=" cell"
          data-testid="cell-5-2"
          role="gridcell"
        />
        <button
          aria-colindex="3"
          aria-selected="false"
          class=" cell"
          data-testid="cell-5-3"
          role="gridcell"
        />
        <button
          aria-colindex="4"
          aria-selected="false"
          class=" cell"
          data-testid="cell-5-4"
          role="gridcell"
        />
        <button
          aria-colindex="5"
          aria-selected="false"
          class=" cell"
          data-testid="cell-5-5"
          role="gridcell"
        />
      </div>
    </div>
  </div>
</div>
`;
